### PR TITLE
Core integration attempt #5

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

**Comment by ry** [comment]
Current stats. About a 2x speed up over the master branch, but still slower than node_tcp.js 

The tail latency (1.4 *seconds* (!)) suggests there's an obvious bug somewhere. So I'm hopeful.

This branch: `target/release/deno tests/http_bench.ts` 
```
~/src/deno> third_party/wrk/mac/wrk -d 10 -c 10 http://127.0.0.1:4500/
Running 10s test @ http://127.0.0.1:4500/
  2 threads and 10 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    23.23ms  114.74ms   1.46s    95.02%
    Req/Sec    18.04k     3.27k   23.84k    75.50%
  359050 requests in 10.00s, 17.46MB read
Requests/sec:  35890.31
Transfer/sec:      1.75MB
```

vs deno v0.3.3 on tests/http_bench.ts
```
~/src/deno> third_party/wrk/mac/wrk -d 10 -c 10 http://127.0.0.1:4500/
Running 10s test @ http://127.0.0.1:4500/
  2 threads and 10 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   546.24us  283.72us  13.45ms   96.30%
    Req/Sec     9.41k     1.08k   10.77k    91.58%
  189045 requests in 10.10s, 9.19MB read
Requests/sec:  18717.73
Transfer/sec:      0.91MB
```

vs `node tools/node_tcp.js`
```
~/src/deno> third_party/wrk/mac/wrk -d 10 -c 10 http://127.0.0.1:4544/
Running 10s test @ http://127.0.0.1:4544/
  2 threads and 10 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   246.54us  259.52us  12.56ms   98.22%
    Req/Sec    20.35k     2.47k   27.66k    76.73%
  408850 requests in 10.10s, 19.89MB read
Requests/sec:  40479.68
Transfer/sec:      1.97MB
```

**Comment by hayd** [comment]
You might like to run `cargo clippy`, it raises a few suggestions (not sure whether they are actually perf issues though).

**Comment by zekth** [comment]
You still bench on raw data or serialized?

File: src/isolate.rs [hunk 1]
```diff
@@ -356,140 +85,77 @@ impl Isolate {
...
     }
     Ok(())
   }
 
-  #[inline]
-  fn ntasks_increment(&self) {
-    assert!(self.ntasks.get() >= 0);
-    self.ntasks.set(self.ntasks.get() + 1);
+  /// Wraps Isolate::mod_new but registers with modules.
+  fn mod_new_and_regsiter(
```
Comment (piscisaureus) [inline src/isolate.rs]: register

File: src/main.rs [hunk 2]
```diff
@@ -1,4 +1,6 @@
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+#![allow(unused_variables)]
```
Comment (piscisaureus) [inline src/main.rs]: Please remove

File: src/isolate.rs (hunk 1 above)
Comment (ry) [reply src/isolate.rs]: fixed

File: src/main.rs (hunk 2 above)
Comment (ry) [reply src/main.rs]: removed

**Pull request merged**